multibuildingdetector/__main__.py

```diff
@@ -1,42 +1,61 @@
-import chainer
 import argparse
-import yaml
-import os
 import importlib
+import os
+import chainer
+import yaml
+import json
 
+from chainer.datasets import TransformDataset
 from chainer.optimizer import WeightDecay
+from chainer.training import extensions
 from chainercv.extensions import DetectionVOCEvaluator
-from chainercv.links import SSD300
 from chainercv.links.model.ssd import GradientScaling
+from os.path import dirname, join, realpath
+
 from multibuildingdetector.transforms.augmentation import ImageAugmentation
-from multibuildingdetector.multiboxtrainchain import MultiboxTrainChain
-from chainer.datasets import TransformDataset
-from chainer.training import extensions
-from .reader import load_train_test_set
+from multibuildingdetector.reader import load_train_test_set
+
+PROJECT_DIR = join(dirname(realpath(__file__)), '..')
 
 
 def load_config(path):
-    with open(path, 'r') as f:
+    with open(os.path.join(PROJECT_DIR, path), 'r') as f:
         return yaml.load(f)
 
 
-def run(input_dir, output, batch_size, train_split=0.8, iterator='SerialIterator',
+def _import_module(package_template, module=''):
+    return importlib.import_module(package_template.format(module))
+
+
+def _import_class(class_path):
+    split = class_path.split('.')
+    class_name = split[-1]
+    package = '.'.join(split[:-1])
+    module = _import_module(package)
+    return getattr(module, class_name)
+
+
+def run(input_dir, output, batch_size, train_split=0.8,
+        iterator='SerialIterator',
         device=-1, pretrained_model='', save_trigger=10000,
-        parser_module='XMLParser'):
+        parser_module='XMLParser',
+        train_module='MultiboxTrainChain',
+        model_module='chainercv.links.SSD300'):
+    pretrained_model = join(PROJECT_DIR, pretrained_model)
     if pretrained_model and os.path.isfile(pretrained_model):
         print('Pretrained model {} loaded.'.format(pretrained_model))
     else:
         print('Pretrained model file not found, ' +
               'using imagenet as default.')
         pretrained_model = 'imagenet'
-    parser = importlib.import_module('multibuildingdetector.parsers.{}'
-                                     .format(parser_module))
-    model = SSD300(n_fg_class=len(parser.LABEL_NAMES),
-                   pretrained_model=pretrained_model)
-    model.use_preset('evaluate')
-    train_chain = MultiboxTrainChain(model)
+    parser = _import_module('multibuildingdetector.parsers.{}', parser_module)
 
-    if device >= 0:
+    model = _import_class(model_module)(n_fg_class=len(parser.LABEL_NAMES),
+                                        pretrained_model=pretrained_model)
+    model.use_preset('evaluate')
+    train_chain = _import_class('multibuildingdetector.trainchains.{}'
+                                .format(train_module))(model)
+    if device > 0:
```

Comment:
please change this back to device >= 0 (0 and 1 are the GPUs, -1 is default for no GPU)